pkg/datapath/linux/node_ids.go

```diff
@@ -126,6 +126,10 @@ func (n *linuxNodeHandler) allocateIDForNode(node *nodeTypes.Node) uint16 {
 
 // deallocateIDForNode deallocates the node ID for the given node, if it was allocated.
 func (n *linuxNodeHandler) deallocateIDForNode(oldNode *nodeTypes.Node) {
+	if len(oldNode.IPAddresses) == 0 {
+		return
+	}
+
 	nodeID := n.nodeIDsByIPs[oldNode.IPAddresses[0].IP.String()]
```

Comment:
Wouldn't the following be a cleaner fix? In symmetry with `allocateIDForNode` above,  `getNodeIDForNode` has a fallback ID of zero if there are no IPs.

```suggestion
	nodeID := n.getNodeIDForNode(oldNode)
```

Reply:
sorry, I didn't notice it, now updated.

Reply:
That function was introduced just yesterday so that might be why you didn't notice when preparing the patch :smile: